sections/todo/time_tracker.js

```diff
@@ -313,35 +316,65 @@ var TimeTracker  = class TimeTracker {
         return false;
     }
 
-    _enable_file_monitors () {
-        [this.daily_csv_file_monitor,] = MISC_UTILS.file_monitor(this.daily_csv_file, () => {
-            this._on_tracker_files_modified();
-        });
-
-        [this.yearly_csv_file_monitor,] = MISC_UTILS.file_monitor(this.yearly_csv_file, () => {
-            this._on_tracker_files_modified();
-        });
-
-        [this.yearly_csv_dir_monitor,] = MISC_UTILS.file_monitor(this.yearly_csv_dir, () => {
-            this._on_tracker_files_modified();
-        });
+    _enable_file_monitors (timeout = 100) {
+        try{
+            if (!this.daily_csv_file_monitor){
+                this.daily_csv_file_monitor = this.daily_csv_file.monitor(Gio.FileMonitorFlags.NONE, null);
+            }
+    
+            if (!this.yearly_csv_file_monitor){
+                this.yearly_csv_file_monitor = this.yearly_csv_file.monitor(Gio.FileMonitorFlags.NONE, null);
+            }
+    
+            if (!this.yearly_csv_dir_monitor){
+                this.yearly_csv_dir_monitor = this.yearly_csv_dir.monitor(Gio.FileMonitorFlags.NONE, null);
+            }
+    
+            Mainloop.timeout_add(timeout,()=>{
+                this.daily_csv_file_monitor_id = this.daily_csv_file_monitor.connect('changed', (...args) =>{
```

Comment:
Is there a reason you removed the `MISC_UTILS.file_monitor` call?

Reply:
Instead of recreating a new file monitor, I think it enough to disconnect the file monitor when changes are made to file. This is not supported by `MISC_UTILS.file_monitor`. But one can probably make some modifications to `MISC_UTILS.file_monitor`.

Reply:
The function returns the monitor_id, so maybe the handler_block function can be used instead of disconnecting.